local-cli/server/util/attachHMRServer.js

```diff
@@ -287,4 +377,28 @@ function time() {
   return `${date.getHours()}:${date.getMinutes()}:${date.getSeconds()}:${date.getMilliseconds()}`;
 }
 
+
+function wrapPotentialPromise (valueOrPromise) {
+  if (valueOrPromise == null) {
+    return Promise.resolve(valueOrPromise)
+  }
+  else if (valueOrPromise.then && typeof valueOrPromise === "function") {
```

Comment:
Definitely! It was a mistake. In practice if the first one is true then it rarely matters but it could be an issue for some promise-likes. This code is being rewritten in a new style so I won't fix it right now. 
Thanks though!